### PR TITLE
Document False usage for old assumptions

### PR DESCRIPTION
**Pull request by skirpichev**
See #8045 and [this question](https://github.com/sympy/sympy/issues/8487#issuecomment-69242730).  I think, there are better source for discussion about False vs None usage, but this is what I've found so far.

### CODE REVIEW

**Comment by pelegm** [comment]
+1

File: sympy/core/assumptions.py [hunk 1]
```diff
@@ -6,8 +6,9 @@
 
 Assumptions determine certain properties of symbolic objects. Assumptions
 can have 3 possible values: True, False, None.  None is returned when it is
-impossible to say something about the property. For example, a generic Symbol
-is not known beforehand to be positive.
+impossible to say something about the property.  For example, a generic Symbol
+is not known beforehand to be positive.  True, if the object has the property
+and False otherwise (i.e. also if the property doesn't make sense for the object).
```
Comment (smichr) [inline sympy/core/assumptions.py]: nit:

can have 3 possible values: True, False, None.  True is returned if the object has the
property (and False is returned if it doesn't). When the property cannot be determined
(or when a method is not implemented) None will be returned, e.g. a generic symbol, x,
may or may not be positive so a value of None is returned for x.is_positive.

OR

+impossible to say something about the property.  (For example, a generic Symbol
+is not known beforehand to be positive.)  False is used when the object either
+doesn't (or can't) have the property and True is returned when the object has the property.

Comment (skirpichev) [inline sympy/core/assumptions.py]: Updated.  I take your first variant with little modifications.

**Comment by asmeurer** [comment]
I find it easiest to think about it in terms of sets. Like for `is_positive` you have the set of positive numbers (say the interval `(0, oo]`). x.is_positive is True if it is known that x is in the set, False if it is known that x is not in the set, and None if it is not known. Not being in the set could be because it is a nonpositive real number or because it is not a number in the first place.

**Comment by asmeurer** [comment]
Even so, what is written here is good and I am +1 to merge it.

**Comment by skirpichev** [comment]
This pr seems to be in a good shape.
Feel free to pick up this work.

**Pull request merged**